logbook-json/src/main/java/org/zalando/logbook/json/JsonPathBodyFilters.java

```diff
@@ -0,0 +1,107 @@
+package org.zalando.logbook.json;
+
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.JsonPath;
+import lombok.RequiredArgsConstructor;
+import net.minidev.json.JSONArray;
+import org.zalando.logbook.BodyFilter;
+
+import javax.annotation.Nullable;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class JsonPathBodyFilters {
+
+    @RequiredArgsConstructor
+    public static class JsonPathBodyFilterBuilder {
+        private final JsonPath jsonPath;
+
+        public BodyFilter delete() {
+            return (JsonBodyFilter) documentContext -> documentContext.delete(jsonPath).jsonString();
+        }
+
+        public BodyFilter replace(String replacement) {
+            return replace(replacement, false, String.class);
+        }
+
+        public BodyFilter forceReplace(String replacement) {
+            return replace(replacement, true, String.class);
+        }
+
+        public BodyFilter replace(Number replacement) {
+            return replace(replacement, false, Number.class);
+        }
+
+        public BodyFilter forceReplace(Number replacement) {
+            return replace(replacement, true, Number.class);
+        }
+
+        public BodyFilter replace(String matchRegex, String replacementRegex) {
+            Pattern pattern = Pattern.compile(matchRegex);
+            return (JsonBodyFilter) documentContext -> {
+                Object obj = documentContext.read(jsonPath);
+                if (isPrimitive(obj)) {
```

Comment:
Could this be simplified by enabling the `ALWAYS_RETURN_LIST` option?

Reply:
The problem with `ALWAYS_RETURN_LIST` is that it is usually used only if we need to parse some data, but not to put it back to document context.
Here's an example:
```
String json = "{\"test\": \"value\"}";
DocumentContext context = JsonPath.using(Configuration.builder().options(Option.ALWAYS_RETURN_LIST).build()).parse(json);
JsonPath path = JsonPath.compile("$.test");
Object read = context.read(path);
System.out.println(read.getClass());
context.set(path, read);
System.out.println(context.jsonString());
```
This code prints:
```
class net.minidev.json.JSONArray
{"test":["value"]}
```

As you can see, we replaced string value with array value. It's not the desired behavior, i suppose :)

Reply:
Does JsonPath provide you with a way to find the definitive path for each match?

Reply:
https://github.com/json-path/JsonPath#path-vs-value that would be nice. You would basically use that option to always get a list of definitive paths and replace those individually.

Reply:
Ok, i just didn't understand your idea :) I think all of the logics may be refactored according to it, including https://github.com/zalando/logbook/pull/1022#discussion_r627321116 this comment